src/core/validate-graph.ts

```diff
@@ -60,6 +60,17 @@ export function validatePackageURL(pkg: types.PkgInfo): void {
         );
         break;
 
+      case 'composer':
+      case 'golang':
+      case 'npm':
+      case 'swift':
+        assert(
+          pkg.name ===
+            [purlPkg.namespace, purlPkg.name].filter(Boolean).join('/'),
+          `name and packageURL name do not match`,
+        );
```

Comment:
Minor: that's quite an expensive way to do it (constructing 3 throw-away arrays and executing a closure). What about something more direct?
```suggestion
        assert(
          pkg.name === purlPkg.namespace
            ? `${purlPkg.namespace}/${purlPkg.name}`
            : purlPkg.name,
          `name and packageURL name do not match`,
        );
```

Reply:
thanks, will change to your proposal!